### PR TITLE
Implement parse identifier

### PR DESCRIPTION
**Pull request by boohyunsik**
resolved: #[45]

1. implement parse identifier
2. implement test cases

- [x] Test case

### CODE REVIEW

File: parse/parser.go [hunk 1]
```diff
@@ -180,9 +180,13 @@ func parseExpAsInfix(buf TokenBuffer, exp ast.Expression, pre precedence) (ast.E
 	return nil, nil
 }
 
-// TODO: implement me w/ test cases :-)
 func parseIdentifier(buf TokenBuffer) (ast.Expression, []error) {
-	return nil, nil
+	errs := make([]error, 0)
+	if buf.Peek(CURRENT).Type != Ident {
+		errs = append(errs, errors.New("parseIdentifier() - This is not a identifier"))
```
Comment (zeroFruit) [inline parse/parser.go]: How about printing the current token when errs? 👍

File: parse/parser.go [hunk 2]
```diff
@@ -180,9 +180,13 @@ func parseExpAsInfix(buf TokenBuffer, exp ast.Expression, pre precedence) (ast.E
...
+		return nil, errs
+	}
```
Comment (junbeomlee) [inline parse/parser.go:189]: you need to add 
`token := buf.Read()` and then
`return &ast.Identifier{Value: token.Val}, nil`

File: parse/parser_internal_test.go [hunk 3]
```diff
@@ -273,3 +275,40 @@ func TestExpectNext(t *testing.T) {
 		tokenBuf.Read()
 	}
 }
+
+func TestParseIdentifier(t *testing.T) {
+	tokenBuf := makeMockTokenBuffer()
+	tests := []struct {
+		expected ast.Expression
+	}{
```
Comment (junbeomlee) [inline parse/parser_internal_test.go:289]: Add `expectedErrs []error` to tests struct to check error

File: parse/parser_internal_test.go [hunk 4]
```diff
@@ -273,3 +275,40 @@ func TestExpectNext(t *testing.T) {
...
+			expected: nil,
+		},
+		{
+			expected: nil,
+		},
+	}
+
+	for i, test := range tests {
+		identifier, _ := parseIdentifier(&tokenBuf)
+
```
Comment (junbeomlee) [inline parse/parser_internal_test.go:314]:
```
identifier, _ := parseIdentifier(&tokenBuf)
if (identifier != nil && test.expected != nil && identifier.String() == test.expected.String()) ||
    identifier == nil && test.expected == nil {
       tokenBuf.Read()
} else {
    t.Fatalf("test[%d] - parseIdentifier() wrong result, expected=%s, got=%s",
 		i, test.expected, identifier)
}
```
should be 
```
exp, errs := parseIdentifier(&tokenBuf)

if errs != test.expectedErrs{
    t.Fatalf("")
}

switch ident := exp.(type) {
case nil:
	if ident != test.expected{
             t.Fatal()
        }
case *ast.Identifier:
	if ident.String() != test.expected.String(){
             t.Fatal()
        }
}

tokenBuf.Read()
```

**Review by junbeomlee — APPROVED** [review]
LGTM

**Comment by boohyunsik** [comment]
Plz merge this request.

**Pull request merged**